File: CS210/HR_Lab6.py

def parse_in(cmds, stack):
    for cmd in cmds:
        if cmd[0:4] == "PUSH":
            stack.append(int(cmd[5:]))
        elif not stack:
            continue
        else: stack.pop()
    return stack

#----------------------------#

cmds = []
n = int(input())

for i in range(n):
    cmds.append(input())

s = parse_in(cmds, [])

if not s:
    print("empty")
else: print(max(s))

File: CS210/HR_Lab6.java
import java.util.ArrayDeque;
import java.util.Scanner;

public class HR_Lab6 {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        ArrayDeque<Integer> stack = new ArrayDeque<Integer>();
        int n = Integer.parseInt(in.nextLine());
        String[][] cmds = new String[n][2];

        for (int i = 0; i < n; i++) {
            cmds[i] = in.nextLine().split(" ");
        }

        ArrayDeque<Integer> list = parseIn(cmds, stack);
        if (list.isEmpty()) {
            System.out.println("empty");
        }
        else System.out.printf("%d\n", max(list));
    }

    static ArrayDeque<Integer> parseIn(String[][] cmds, ArrayDeque<Integer> stack) {
        for (String[] cmd : cmds) {
            if (cmd[0].equals("PUSH")) {
                stack.addFirst(Integer.parseInt(cmd[1]));
            }
            else if (stack.isEmpty()) {
                continue;
            }
            else {
                stack.removeFirst();
            }
        }
        return stack;
    }

    static int max(ArrayDeque<Integer> stack) {
        int max = stack.removeFirst();
        while (!stack.isEmpty()) {
            if (stack.peekFirst() > max) {
                max = stack.removeFirst();
            }
            else {
                stack.removeFirst();
            }
        }
        return max;
    }
}
